zipline/utils/cache.py

```diff
@@ -298,9 +304,25 @@ def __enter__(self):
         return self
 
     def __exit__(self, *exc_info):
+        self._tmpfile.__exit__(*exc_info)
         if exc_info[0] is None:
             self._commit()
-        self._tmpfile.__exit__(*exc_info)
+
+    @property
```

Comment:
if we need to handle that in the future then we can swap to a property. I think that for now it is more clear to just expose the attribute.